File: androidApp/src/main/java/com/cocktailcraft/screens/CartScreen.kt
package com.cocktailcraft.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.itemsIndexed
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.ShoppingCart
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.cocktailcraft.ui.components.CartItemCard
import com.cocktailcraft.ui.components.ConfirmationDialog
import com.cocktailcraft.ui.components.EmptyStateComponent
import com.cocktailcraft.ui.components.LoadingStateComponent
import com.cocktailcraft.ui.components.OrderSummaryCard
import com.cocktailcraft.ui.components.SectionHeader
import com.cocktailcraft.ui.theme.AppColors
import com.cocktailcraft.viewmodel.CartViewModel
import com.cocktailcraft.viewmodel.FavoritesViewModel
import com.cocktailcraft.viewmodel.OrderViewModel
import com.cocktailcraft.navigation.NavigationManager
import java.text.NumberFormat
import java.util.Locale

@Composable
fun CartScreen(
    viewModel: CartViewModel,
    orderViewModel: OrderViewModel,
    navigationManager: NavigationManager,
    onStartShopping: () -> Unit,
    favoritesViewModel: FavoritesViewModel
) {
    val cartItems by viewModel.cartItems.collectAsState()
    val totalPrice by viewModel.totalPrice.collectAsState()
    val isLoading by viewModel.isLoading.collectAsState()
    val error by viewModel.error.collectAsState()
    val favorites by favoritesViewModel.favorites.collectAsState()

    val currencyFormatter = NumberFormat.getCurrencyInstance(Locale.US)

    var showPlaceOrderDialog by remember { mutableStateOf(false) }

    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(AppColors.Background)
            .padding(16.dp)
    ) {
        // Show loading state
        LoadingStateComponent(isLoading = isLoading)

        // Show error state
        if (!isLoading && error != null) {
            EmptyStateComponent(
                title = "Error",
                message = error ?: "An unknown error occurred",
                actionButtonText = "Try Again",
                onActionButtonClick = { /* Add retry logic here */ }
            )
        }
        // Show empty cart state
        else if (!isLoading && cartItems.isEmpty()) {
            EmptyStateComponent(
                title = "Your cart is empty",
                message = "Add some cocktails to your cart and they will appear here",
                actionButtonText = "Start Shopping",
                onActionButtonClick = onStartShopping,
                icon = Icons.Filled.ShoppingCart
            )
        }
        // Show cart items
        else if (!isLoading) {
            SectionHeader(
                title = "Your Cart",
                fontSize = 20,
                modifier = Modifier.padding(bottom = 8.dp)
            )

            LazyColumn(
                modifier = Modifier
                    .weight(1f)
                    .fillMaxWidth(),
                verticalArrangement = Arrangement.spacedBy(12.dp),
                contentPadding = PaddingValues(bottom = 8.dp)
            ) {
                itemsIndexed(cartItems) { _, item ->
                    CartItemCard(
                        item = item,
                        onIncreaseQuantity = { viewModel.updateQuantity(item.cocktail.id, item.quantity + 1) },
                        onDecreaseQuantity = {
                            if (item.quantity > 1) {
                                viewModel.updateQuantity(item.cocktail.id, item.quantity - 1)
                            } else {
                                viewModel.removeFromCart(item.cocktail.id)
                            }
                        },
                        onRemove = { viewModel.removeFromCart(item.cocktail.id) },
                        isFavorite = favorites.any { it.id == item.cocktail.id },
                        onToggleFavorite = { favoritesViewModel.toggleFavorite(item.cocktail) }
                    )
                }
            }

            Spacer(modifier = Modifier.height(16.dp))

            // Order Summary
            OrderSummaryCard(
                subtotal = totalPrice,
                deliveryFee = 5.99,
                modifier = Modifier.padding(vertical = 8.dp),
                currencyFormatter = currencyFormatter
            )

            Spacer(modifier = Modifier.height(16.dp))

            // Checkout Button
            Button(
                onClick = { showPlaceOrderDialog = true },
                modifier = Modifier
                    .fillMaxWidth()
                    .height(54.dp),
                colors = ButtonDefaults.buttonColors(
                    containerColor = AppColors.Primary
                ),
                shape = RoundedCornerShape(12.dp)
            ) {
                Text(
                    text = "Place Order",
                    fontWeight = FontWeight.Bold,
                    fontSize = 16.sp
                )
            }
        }
    }

    // Order Confirmation Dialog
    ConfirmationDialog(
        showDialog = showPlaceOrderDialog,
        title = "Confirm Order",
        message = "Are you sure you want to place this order for ${currencyFormatter.format(totalPrice + 5.99)}?",
        confirmButtonText = "Confirm",
        dismissButtonText = "Cancel",
        onConfirm = {
            orderViewModel.placeOrder(cartItems, totalPrice)
            viewModel.clearCart()
            showPlaceOrderDialog = false
            navigationManager.navigateToOrderList()
        },
        onDismiss = { showPlaceOrderDialog = false }
    )
}

File: iosApp/CocktailCraft/CocktailCraft/Screens/CartScreen.swift
import SwiftUI
import shared

struct CartScreen: View {
    @ObservedObject private var cartViewModel = ViewModelProvider.shared.cartViewModel
    @ObservedObject private var orderViewModel = ViewModelProvider.shared.orderViewModel
    @ObservedObject private var favoritesViewModel = ViewModelProvider.shared.favoritesViewModel
    @EnvironmentObject var navigationCoordinator: NavigationCoordinator
    
    @State private var showPlaceOrderAlert = false
    
    private var currencyFormatter: NumberFormatter {
        let formatter = NumberFormatter()
        formatter.numberStyle = .currency
        formatter.locale = Locale(identifier: "en_US")
        return formatter
    }
    
    var body: some View {
        NavigationView {
            ZStack {
                if cartViewModel.isLoading {
                    ProgressView("Loading cart...")
                        .frame(maxWidth: .infinity, maxHeight: .infinity)
                } else if let error = cartViewModel.errorString {
                    EmptyStateView(
                        title: "Error",
                        message: error,
                        systemImage: "exclamationmark.triangle",
                        actionButtonText: "Try Again",
                        onActionButtonClick: {
                            // Cart doesn't have refresh, reload from parent view
                        }
                    )
                } else if cartViewModel.cartItems.isEmpty {
                    EmptyStateView(
                        title: "Your cart is empty",
                        message: "Add some cocktails to your cart and they will appear here",
                        systemImage: "cart",
                        actionButtonText: "Start Shopping",
                        onActionButtonClick: {
                            navigationCoordinator.selectedTab = .home
                        }
                    )
                } else {
                    VStack(spacing: 0) {
                        // Cart Items List
                        ScrollView {
                            LazyVStack(spacing: 12) {
                                ForEach(cartViewModel.cartItems, id: \.cocktail.id) { item in
                                    CartItemView(
                                        item: item,
                                        isFavorite: favoritesViewModel.favorites.contains { $0.id == item.cocktail.id },
                                        onIncreaseQuantity: {
                                            cartViewModel.updateQuantity(
                                                cocktailId: item.cocktail.id,
                                                quantity: item.quantity + 1
                                            )
                                        },
                                        onDecreaseQuantity: {
                                            if item.quantity > 1 {
                                                cartViewModel.updateQuantity(
                                                    cocktailId: item.cocktail.id,
                                                    quantity: item.quantity - 1
                                                )
                                            } else {
                                                cartViewModel.removeFromCart(cocktailId: item.cocktail.id)
                                            }
                                        },
                                        onRemove: {
                                            cartViewModel.removeFromCart(cocktailId: item.cocktail.id)
                                        },
                                        onToggleFavorite: {
                                            favoritesViewModel.toggleFavorite(cocktail: item.cocktail)
                                        }
                                    )
                                }
                            }
                            .padding()
                        }
                        
                        // Bottom Section with Order Summary and Checkout Button
                        VStack(spacing: 16) {
                            OrderSummaryView(
                                subtotal: cartViewModel.totalPrice,
                                deliveryFee: 5.99,
                                total: nil,
                                showDeliveryFee: true,
                                additionalItems: nil
                            )
                            
                            Button(action: {
                                showPlaceOrderAlert = true
                            }) {
                                Text("Place Order")
                                    .font(.system(size: 16, weight: .bold))
                                    .foregroundColor(.white)
                                    .frame(maxWidth: .infinity)
                                    .frame(height: 54)
                                    .background(Color.blue)
                                    .cornerRadius(12)
                            }
                        }
                        .padding()
                        .background(Color(UIColor.systemBackground))
                        .shadow(color: Color.black.opacity(0.05), radius: 8, x: 0, y: -2)
                    }
                }
            }
            .navigationTitle("Cart")
            .navigationBarTitleDisplayMode(.large)
            .confirmationAlert(
                isPresented: $showPlaceOrderAlert,
                title: "Confirm Order",
                message: "Are you sure you want to place this order for \(currencyFormatter.string(from: NSNumber(value: cartViewModel.totalPrice + 5.99)) ?? "$0.00")?",
                confirmButtonText: "Confirm",
                onConfirm: {
                    placeOrder()
                }
            )
        }
    }
    
    private func placeOrder() {
        orderViewModel.placeOrder(
            items: cartViewModel.cartItems,
            totalPrice: cartViewModel.totalPrice
        )
        cartViewModel.clearCart()
        // Navigate to orders tab
        navigationCoordinator.selectedTab = .orders
    }
}